src-local/ca_continuation_utils.py

```
@@ -0,0 +1,647 @@
+        interval_width = Ca_fail - Ca_good
+        
+        # Check convergence
+        if interval_width < tolerance:
+            print(f"Converged to Ca_cr = {Ca_good:.{decimal_places}f} after {iter_count} iterations")
+            break
+            
+        # Special handling when interval is very small but not converged
+        # This happens when Ca_critical is extremely close to Ca_good
+        if interval_width < 100 * tolerance and consecutive_bisection_failures > 5:
+            print(f"\nInterval very small ({interval_width:.2e}) with many failures.")
+            print(f"Switching to logarithmic search in tiny interval...")
```

Ruff (0.11.9)

f-string without any placeholders

Remove extraneous `f` prefix

(F541)